e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,160 @@
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	By("bootstrapping test environment")
+	useExistingCluster := true
+	testEnv = &envtest.Environment{
+		UseExistingCluster: &useExistingCluster,
+	}
+
+	var err error
+	// cfg is defined in this file globally.
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+})
+
+var _ = AfterSuite(func() {
+	By("tearing down the test environment")
+	err := testEnv.Stop()
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = Describe("e2e test", func() {
+
+	It("testing integrity", func() {
+		By("new file added")
+		k8sClient, err := kubernetes.NewForConfig(cfg)
+		Expect(err).ToNot(HaveOccurred(), "failed to get client")
+		podName, podNamespace, err := GetPodName(k8sClient)
+		fmt.Println("pod spec", podName, podNamespace)
+		Expect(err).ToNot(HaveOccurred())
+		cmd := exec.Command("kubectl", "-n", podNamespace, "exec", "-it", podName, "--", "touch", "/usr/bin/newfile")
+		Expect(cmd.Start()).To(Succeed())
+
+		// give a time to check files
+		time.Sleep(2 * time.Second)
+		getLogs := exec.Command("kubectl", "-n", podNamespace, "logs", "svc/rsyslog")
+		out, err := getLogs.Output()
+		Expect(err).ToNot(HaveOccurred(), "failed to get logs")
+		msg := GetLastMsg(out)
+		fmt.Println(msg)
+		expectRegexp := "\\d*-\\d*-\\d*T\\d*:\\d*:\\d*\\+\\d*:\\d*\\sapp-nginx-integrity\\.default\\sintegrity-monitor\\[\\d*\\]:\\stime=\\w*\\s\\d*\\s\\d*:\\d*:\\d*\\sevent-type=\\d*\\sservice=nginx\\spod=app-nginx-integrity-\\w*-\\w*\\simage=nginx:\\d*.\\d*.\\d*\\snamespace=default\\scluster=local\\smessage=Restart pod app-nginx-integrity-\\w*-\\w*\\sfile=usr\\/bin\\/newfile\\sreason=new\\sfile\\sfound"
+		Expect(msg).To(MatchRegexp(expectRegexp))
+
+		By("file deleted")
+		// waiting pod restart
+		time.Sleep(5 * time.Second)
+		k8sClient, err = kubernetes.NewForConfig(cfg)
+		Expect(err).ToNot(HaveOccurred(), "failed to get client")
+		podName, podNamespace, err = GetPodName(k8sClient)
+		Expect(err).ToNot(HaveOccurred())
+		cmd = exec.Command("kubectl", "-n", podNamespace, "exec", "-it", podName, "--", "rm", "-f", "/usr/bin/tr")
+		Expect(cmd.Start()).To(Succeed())
+
+		// give a time to check files
+		time.Sleep(2 * time.Second)
+		getLogs = exec.Command("kubectl", "-n", podNamespace, "logs", "svc/rsyslog")
+		out, err = getLogs.Output()
+		Expect(err).ToNot(HaveOccurred(), "failed to get logs")
+		msg = GetLastMsg(out)
+		fmt.Println(msg)
+		expectRegexp = "reason=file deleted"
+		Expect(msg).To(MatchRegexp(expectRegexp))
+
+		By("file changed")
+		// waiting pod restart
+		time.Sleep(5 * time.Second)
+		k8sClient, err = kubernetes.NewForConfig(cfg)
+		Expect(err).ToNot(HaveOccurred(), "failed to get client")
+		podName, podNamespace, err = GetPodName(k8sClient)
+		Expect(err).ToNot(HaveOccurred())
+		cmd = exec.Command("kubectl", "-n", podNamespace, "exec", "-it", podName, "--", "cp", "/usr/bin/cut", "/usr/bin/tr")
```

Comment:
Sorry, but it is not clear what are the purposes for the files `/usr/bin/cut`, `/usr/bin/tr`, `/usr/bin/newfile`